packages/gestalt/src/Avatar.js

```diff
@@ -84,10 +85,12 @@ export default class Avatar extends React.PureComponent<AvatarProps, State> {
   handleImageError = () => this.setState({ isImageLoaded: false });
 
   render() {
-    const { name, outline, size, src, verified } = this.props;
+    const { name, outline, size, src, verified, icon } = this.props;
     const { isImageLoaded } = this.state;
     const width = size ? sizes[size] : '100%';
     const height = size ? sizes[size] : '';
+    const verifiedIcon = icon && icon === 'pinterest' ? icon : 'check-circle';
```

Comment:
i dont think you need this logic, you can just have `icon={icon}` down below then set `check-circle` as the default prop for the icon value since that prop is only ever used if verified is true anyways.

another route is just `icon || 'check-circle'` but i prefer the first way of default props